returnn/util/file_cache.py

```diff
@@ -327,6 +327,20 @@ def _check_existing_copied_file_maybe_cleanup(src_filename: str, dst_filename: s
         return True
 
 
+def _copy_with_prealloc(src: str, dst: str):
+    """
+    copies from src to dst preallocating the disk space at dst before copying
+    :param src: source file
+    :param dst: destination file
+    """
+    stat = os.stat(src)
+    with open(dst, "wb") as dst_file:
+        dst_file.truncate(stat.st_size)
```

Comment:
I'm not 100% sure this will always really allocate the space. Some filesystems could support sparse files? Actually maybe most?

Note, I have also see other possibilities. You should also add some comment shortly describing the possible options for the preallocation.

E.g.:

* `os.seek(size-2)` + `file.write(b"\0")` + `os.seek(0)`. ([via](https://stackoverflow.com/a/68104312/133374))
* `os.posix_fallocate`: Probably the best option, as it is specifically intended for this.